src/libopensc/card-gids.c

```diff
@@ -691,7 +691,8 @@ static int gids_finish(sc_card_t *card)
 // the cmap file is used to detect the key algorithm / size
 static int gids_get_crypto_identifier_from_key_ref(sc_card_t *card, const unsigned char keyref, unsigned char *cryptoidentifier) {
 	struct gids_private_data *data = (struct gids_private_data *) card->drv_data;
-	PCONTAINER_MAP_RECORD records = (PCONTAINER_MAP_RECORD) data->cmapfile;
+	PCONTAINER_MAP_RECORD records = malloc(sizeof(CONTAINER_MAP_RECORD));
```

Comment:
this is the error:
```
card-gids.c:694:2: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
  PCONTAINER_MAP_RECORD records = (PCONTAINER_MAP_RECORD) data->cmapfile;
  ^
cc1: all warnings being treated as errors
```

Reply:
The fix is not correct as the `data->cmapfile` can contain multiple `CONTAINER_MAP_RECORD` records if I read right, but you malloc & copy just one.

But right now, I do not see a simple way how to fix this. I read this assignment from `u8` ("alias" to `unsigned char` in types.h) `cmapfile` to `PCONTAINER_MAP_RECORD` valid. The other option I see suggested would be use of union to reference same data with different types, but I think it would require static size.

Reply:
Ack, any hints how to solve this appreciated. I'll push this patch to another PR though, so that
we can fix the maybe uninitialized warnings treated as errors.